griptape/schemas/union_field.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+from typing import Any
+
+import marshmallow
+import marshmallow.error_store
+import marshmallow.exceptions
+
+
+class MarshmallowUnionError(Exception):
+    """Base error for marshmallow_union."""
+
+
+class ExceptionGroupError(MarshmallowUnionError):
+    """Collection of possibly multiple exceptions."""
+
+    def __init__(self, msg: str, errors: Any) -> None:
+        self.msg = msg
+        self.errors = errors
+        super().__init__(msg, errors)
+
+
+class UnionField(marshmallow.fields.Field):
```

Comment:
Class should still be called `Union` for consistency with `Bytes`. When importing elsewhere you can introduce an alias `from griptape.schemas import Union as UnionField`